src/java.base/share/classes/java/lang/constant/ClassDesc.java

```diff
@@ -103,7 +109,7 @@ static ClassDesc of(String name) {
      * @since 20
      */
     static ClassDesc ofInternalName(String name) {
-        ConstantUtils.validateInternalClassName(requireNonNull(name));
+        validateInternalClassName(requireNonNull(name));
         return ClassDesc.ofDescriptor("L" + name + ";");
```

Comment:
We can use `ofTrusted` here and above if we validate the binary/internal names to have no trailing or consecutive slash/dots.